packages/pieces/community/pinterest/src/lib/actions/delete-pin.ts

```diff
@@ -0,0 +1,23 @@
+import { createAction, Property } from '@activepieces/pieces-framework';
+import { makeRequest } from '../common';
+import { pinterestAuth } from '../common/auth';
+import { HttpMethod } from '@activepieces/pieces-common';
+import { pinIdDropdown } from '../common/props';
+
+export const deletePin = createAction({
+  auth: pinterestAuth,
+  name: 'deletePin',
+  displayName: 'Delete Pin',
+  description: 'Permanently delete a specific Pin.',
+  props: {
+    pin_id: pinIdDropdown,
+  },
+  async run({ auth, propsValue }) {
+    const { pin_id } = propsValue;
+    return await makeRequest(
```

Comment:
Response Handling: The API returns 204 (No Content) on success, but the current implementation might not handle this properly. Consider adding success confirmation:
```suggestion
       async run({ auth, propsValue }) {
     const { pin_id, ad_account_id } = propsValue;
     
     let path = `/pins/${pin_id}`;
     if (ad_account_id) {
       path = `/pins/${pin_id}?ad_account_id=${ad_account_id}`;
     }
     
     const response = await makeRequest(
       auth.access_token as string,
       HttpMethod.DELETE,
       path
     );
     
     return {
       success: true,
       message: `Pin ${pin_id} has been successfully deleted.`,
       pin_id: pin_id
     };
   }
```